### PR TITLE
DE5386: Introduce PFI debounce (0.5ms)

### PR DESCRIPTION
**Pull request by chadbarth**
Trust me (or not).

### CODE REVIEW

File: pfi.c [hunk 1]
```diff
@@ -116,11 +116,19 @@ static int __init pfi_init(void)
...
 
     result = gpio_request_one(pin, GPIOF_IN, DRV_NAME " irq");
     if (result >= 0)
     {
+        // Set debounce to 500 microseconds. This actually sets it for the entire
+        // gpio bank, not just the specific pin.
+        result = gpio_set_debounce(pin, 500);
+        if(result != 0)
+        {
+            printk(KERN_ALERT DRV_NAME ": gpi_set_debounce returned %d", result);
```
Comment (chrisko) [inline pfi.c]: Could add a `\n`, I guess.

**Review by chrisko — APPROVED** [review]
LGTM! I must be a trusting sort of guy.

Did Kris tell us anything about why he needs this? I'm curious.

File: pfi.c (hunk 1 above)
Comment (chadbarth) [reply pfi.c]: I guess it could. Funny, I didn't really notice anything wrong with it in /var/log/messages.

**Comment by chadbarth** [comment]
@chrisko: Kris was seeing some failures in EMC testing. With certain levels of noise in the power supply, the device would suddenly go into "Recovery" and then, after a short time (order of 1 minute), the device would reboot. He traced it to a momentary blip on the power fail line, somewhere in the order of 10ns. The kernel module, and XL process would see the blip and start shutting down. We set a 1 minute timer in this case so we don't just put ourselves to sleep if the power doesn't actually fail. 

Kris asked me to look at some hardware registers, which I was able to trace up to this `gpio_set_debounce` kernel  function.

But, this all just proves that you don't trust me. ;-)

File: pfi.c (hunk 1 above)
Comment (chadbarth) [reply pfi.c]: See https://github.com/Vorne/pfi/pull/5/commits/32939cb76e9cc12989466e9b6471c79bdb437360. The module was also rebuilt.

**Comment by chrisko** [comment]
Trust... but verify :p

Thanks for the deets! Hardware, man.

**Comment by robgura** [comment]
Should we check in the linux headers that you used?

**Comment by chadbarth** [comment]
@rob: The "headers" I have are in sort of half baked state. The .deb archive I built has some tools that are not properly cross-compiled, so I want to look into fixing that (I just copied over working versions of the tools temporarily to get the object built).

But, long term, yes. I do want to capture the stuff we depend on somewhere, I was just planning to do that elsewhere (or at least in another PR).

**Pull request merged**